### DISCUSSION TITLE
Replace deprecated method getProvider with the new one getComponentProvider.

### DISCUSSION THREAD

**Discussion by mcardia** [keycloak/keycloak#16049] [Q&A]
How do I replace KeycloakSession.getProvider with KeycloakSession.getComponentProvider ?
In the exemplo below, model is the _UserStorageProviderModel_ from sync/syncSince methods from _UserStorageProviderFactory_

Old method works:
`UserStorageProvider provider = keycloakSession.getProvider(UserStorageProvider.class, this.model);`

New method returns **null** (I tried, with model.getId(), model.getProviderId() and model.getName())
`UserStorageProvider provider = keycloakSession.getComponentProvider(UserStorageProvider.class, this.model.getProviderId());`

Thanks

**Comment by nxglhez** [comment]
And sadly in https://github.com/keycloak/keycloak/blob/296e1a5ccc0dffb3f8cf87d42252e2c9dd20b2b7/federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java#L636 it use the deprecated method. 

Your problem here is that when you use `getComponentProvider` it will use `create(KeycloakSession)` whose default method returns `null`.

I don't know what is expected to be implemented:

- it seems that your factory will receive the `ComponentModel` in the factory' init method as `ComponentModelScope`.
- if you are trying to implements `ImportSynchronization`, you already have the `ComponentModel`
